csrfguard/src/main/java/org/owasp/csrfguard/config/PropertiesConfigurationProvider.java

```diff
@@ -563,13 +568,28 @@ private void javascriptInitParamsIfNeeded() {
 				} else if (servletConfig.getServletContext().getRealPath(this.javascriptSourceFile) != null) {
 					this.javascriptTemplateCode = CsrfGuardUtils.readFileContent(servletConfig.getServletContext().getRealPath(this.javascriptSourceFile));
 				} else {
-					throw new IllegalStateException("getRealPath failed for file " + this.javascriptSourceFile);
+					try( final InputStream inputStream = getResourceStream(this.javascriptSourceFile, servletConfig)){
+						this.javascriptTemplateCode = CsrfGuardUtils.readInputStreamContent(inputStream);	
+					} catch (final Exception e) {
+						throw new IllegalStateException("getRealPath failed for file " + this.javascriptSourceFile);
+					}
 				}
 
 				this.javascriptParamsInitialized = true;
 			}
 		}
 	}
+	
+	private InputStream getResourceStream(final String resourcePath, final ServletConfig servletConfig) throws MalformedURLException  {
```

Comment:
getResourceStream() will return inputstream if resource is present at that path or else if will return null.